### PR TITLE
Send email preview

### PR DESCRIPTION
**Pull request by richardolsson**
## Description
This PR implements UI to interact with the new preview API (still not deployed on the backend), which the user can use to send an email to themselves to preview it.

## Screenshots
None

## Changes
* Fixes some mistakes with types of existing features
* Implement real sending in the `useSendTestEmail()` hook
* Tweak preview UI to display a message when the email has been sent
* Use styling from frame in editor and implement logic to rewrite CSS for this to work

## Notes to reviewer
The API for this has not yet been deployed on the dev server, so trying to send will likely result in an error response (404) from the API.

## Related issues
Undocumented

### CODE REVIEW

File: src/features/emails/hooks/useSendTestEmail.ts [hunk 1]
```diff
@@ -1,19 +1,42 @@
 import { useState } from 'react';
 
+import { useUser } from 'utils/hooks/useFocusDate';
+import { useApiClient, useNumericRouteParams } from 'core/hooks';
+
 export default function useSendTestEmail() {
+  const apiClient = useApiClient();
   const [isLoading, setIsLoading] = useState(false);
+  const [isPostSend, setIsPostSend] = useState(false);
```
Comment (ziggabyte) [inline src/features/emails/hooks/useSendTestEmail.ts:9]: I wish this state was named differently, I have tried for a while and I don't understand what this means - thinking either that PostSend is a type of object or something and this says if something is a PostSend or not, in that case maybe call it isCorrectType or something? OR post here means after, and in that case maybe call it wasSent or isAfterSending or something?

**Review by ziggabyte — APPROVED** [review]
This looks nice to me :D Just one small comment on the name of a state, it's not crucial for functionality so will approve and you can choose to address it if you want.

File: src/features/emails/hooks/useSendTestEmail.ts (hunk 1 above)
Comment (richardolsson) [reply src/features/emails/hooks/useSendTestEmail.ts]: You're right! I renamed it to `emailWasSent`.

**Comment by richardolsson** [comment]
I'm daring to merge since the only change since the review was the requested change, and all checks are passing.

**Pull request merged**